addon/modules/object-browser-serializer.js

```diff
@@ -0,0 +1,198 @@
+import Ember from 'ember'
+const {on} = Ember
+import offsetPagination from './pagination/offset'
+
+export default Ember.Object.extend({
+  initContext: on('init', function () {
+    /* eslint-disable */
+    Ember.defineProperty(this, 'pagination', undefined, (function () {
+      let pagination = this.get('config.page')
+      if (typeof pagination === 'undefined') {
+        return offsetPagination
+      } else {
+        return this.get('pagination.strategy')
+      }
+    }).call(this))
+    /* eslint-enable */
+  }),
+
+  // normalization methods
+  normalizeFilter: function (filter) {
+    // normalize processing
+    let keys = Object.keys(filter)
+    if (!Ember.isPresent(keys)) {
+      return []
+    }
+    // get rid of junks prop from component output
+    let processedFilter = {}
+    keys.forEach((key) => {
+      processedFilter[key] = filter[key]
+    })
+    return processedFilter
+  },
+
+  normalizeSort: function (sort) {
+    if (!Ember.isPresent(sort)) {
+      return []
+    }
+    return sort.map(function (item) {
+      let key = item.value
+      let direction = item.direction === ':desc' ? '-' : ''
+      return `${direction}${key}`
+    })
+  },
+
+  normalizePage: function (page) {
+    return page
+  },
+
+  // hooks that can be overwritten
+  serializeQueryParams: function (queryObject) {
+    return queryObject
+  },
+
+  didReceiveResponse: function (response) {
+    return response
+  },
+
+  willQuery: function () {
```

Comment:
Are we planning to use the empty functions? Should we remove them since we are not using them?